groth16/src/prover.rs

```diff
@@ -234,7 +239,8 @@ where
     end_timer!(synthesis_time);
 
     let witness_map_time = start_timer!(|| "R1CS to QAP witness map");
-    let (full_input_assignment, h, _) = R1CStoQAP::witness_map::<E>(&prover, &d1, &d2, &d3)?;
+    let (full_input_assignment, h, _) =
```

Comment:
if d1, d2 and d3 are always zero, why do we need to pass them in here?

Reply:
This is a witness_map that also works for PGHR13. I kept it for the possible future where someone wants that, but maybe we can remove it :) Wdyt?

Reply:
I think we can remove it, I feel like QAP based SNARKs are going out of fashion haha

Reply:
I think it was @gakonst who gave a presentation about "Groth16 is not dead yet". Maybe PGHR13 is though :P I'll remove it.

Reply:
OK, done!